tests/unit/test_parameters.py

```diff
@@ -0,0 +1,16 @@
+from databricks.sql.utils import named_parameters_to_tsparkparams
+from databricks.sql.thrift_api.TCLIService.ttypes import (
+    TSparkParameter,
+    TSparkParameterValue
+)
+from databricks.sql.utils import DbSqlParameter, DbSqlType
+
+class TestTSparkParameterConversion(object):
```

Comment:
We need to split this into some smaller tests that isolate the different behaviours. The test is currently hard to read and is also missing a test check for type inference of a `None` value.

Ideally we'd like to test `infer_types` directly rather than the higher-up function `named_parameters_to_tsparkparams` so we can more directly verify each step of the way.

Reply:
I think that the sql void type is unsupported.

Reply:
Unsupported by what? VOID is a valid data type on [Databricks](https://docs.databricks.com/en/sql/language-manual/sql-ref-datatypes.html).

Even if it's not supported for parameterized execution, we need to capture the connector behaviour when a user passes a Python `None` as a parameter value. At minimum an exception should be raised and we should have a test to prove that this happens.

Reply:
Ok, we can definitely raise an exception. Just wanted to point out that across all connectors we are not supporting the void type for parameterized exec.

Reply:
This now should raise a value exception if none is passed in.